packages/jest-util/src/createProcessObject.ts

```diff
@@ -8,49 +8,60 @@
 import deepCyclicCopy from './deepCyclicCopy';
 
 const BLACKLIST = new Set(['env', 'mainModule', '_events']);
+const isWin32 = process.platform === 'win32';
+const proto: Record<string, any> = Object.getPrototypeOf(process.env);
 
 // The "process.env" object has a bunch of particularities: first, it does not
 // directly extend from Object; second, it converts any assigned value to a
 // string; and third, it is case-insensitive in Windows. We use a proxy here to
 // mimic it (see https://nodejs.org/api/process.html#process_process_env).
 
 function createProcessEnv(): NodeJS.ProcessEnv {
-  if (typeof Proxy === 'undefined') {
```

Comment:
Not needed?

Reply:
Node.js 6 supports Proxy. 
I think it was old polyfill for Node 5 and below.